Test/CoreSDK.Test/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModuleTest.cs

```diff
@@ -53,20 +56,47 @@ public void TestDiagnosticsModuleSetSeverity()
             {
                 initializedModule.Initialize(new TelemetryConfiguration());
                 
-                Assert.AreEqual(EventLevel.Error.ToString(), initializedModule.Severity);
+                Assert.Equal(EventLevel.Error.ToString(), initializedModule.Severity);
 
                 initializedModule.Severity = "Informational";
 
-                Assert.AreEqual(EventLevel.Informational, initializedModule.EventListener.LogLevel);
+                Assert.Equal(EventLevel.Informational, initializedModule.EventListener.LogLevel);
             }
         }
 
         [TestMethod]
         public void TestDiagnosticModuleDoesNotThrowIfInitailizedTwice()
         {
-            DiagnosticsTelemetryModule module = new DiagnosticsTelemetryModule();
-            module.Initialize(TelemetryConfiguration.Active);
-            module.Initialize(TelemetryConfiguration.Active);
+            using (DiagnosticsTelemetryModule module = new DiagnosticsTelemetryModule())
+            {
+                module.Initialize(new TelemetryConfiguration());
+                module.Initialize(new TelemetryConfiguration());
+            }
+        }
+
+        [TestMethod]
+        public void DiagnosticModuleDoesNotThrowIfQueueSenderContinuesRecieveEvents()
+        {
+            using (DiagnosticsTelemetryModule module = new DiagnosticsTelemetryModule())
+            {
+                var queueSender = module.Senders.OfType<PortalDiagnosticsQueueSender>().First();
+
+                CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+                TaskEx.Run(() =>
+                {
+                    while (!cancellationTokenSource.IsCancellationRequested)
+                    {
+                        queueSender.Send(new TraceEvent());
+                        Thread.Sleep(1);
+                    }
+                }, cancellationTokenSource.Token);
+
+
+                Assert.DoesNotThrow(() => module.Initialize(new TelemetryConfiguration()));
+
+                cancellationTokenSource.Cancel();
+                cancellationTokenSource.Dispose();
```

Comment:
How about wrapping the cancellationTokenSource in a using that would call Dispose automatically?